brjs-core-tests/src/test-integration/java/org/bladerunnerjs/api/memoization/FileObserverTest.java

```diff
@@ -0,0 +1,300 @@
+package org.bladerunnerjs.api.memoization;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.commons.io.FileUtils;
+import org.bladerunnerjs.api.BRJS;
+import org.bladerunnerjs.api.FileObserver;
+import org.bladerunnerjs.api.FileObserverMessages;
+import org.bladerunnerjs.api.spec.engine.SpecTest;
+import org.bladerunnerjs.memoization.PollingFileModificationObserver;
+import org.bladerunnerjs.memoization.WatchingFileModificationObserver;
+import org.bladerunnerjs.spec.brjs.BRJSTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+
+@RunWith(Parameterized.class)
+public class FileObserverTest extends SpecTest
+{
+	
+	private FileObserverFactory fileObserverFactory;
+	private FileObserver fileObserver;
+	private FileModificationRegistry modificationRegistry;
+	private File brjsDir;
+	private File secondaryTempFolder;
+	
+	@Parameters(name="{0}")
+	public static Collection<Object[]> getParameters() {
+		FileObserverFactory pollingObserverFactory = new FileObserverFactory()
+		{
+			public FileObserver createObserver(BRJS brjs)
+			{
+				return new PollingFileModificationObserver(brjs, 100);
+			}
+		};
+		FileObserverFactory watchingObserverFactory = new FileObserverFactory()
+		{
+			public FileObserver createObserver(BRJS brjs)
+			{
+				return new WatchingFileModificationObserver(brjs);
+			}
+		};
+		
+		return Arrays.asList(new Object[][]{
+				{"Polling", pollingObserverFactory},
+				{"Watching", watchingObserverFactory}
+			});
+	}
+	
+	public FileObserverTest(String testName, FileObserverFactory fileObserverFactory) {
```

Comment:
It's used via reflection. The first arg on line 54 and 55 are the name for the parameterized test.